### PR TITLE
fix: ensure abe2e SIG image replication

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

**What this PR does / why we need it**:

adds new logic to abe2es to ensure that the selected VHDs to run the various scenarios are replicated to the particular region in which the suite is being run

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: e2e/config/vhd.go [hunk 1]
```diff
@@ -114,29 +185,58 @@ func findLatestImageWithTag(imageID, tagName, tagValue string) (VHDResourceID, e
...
-	return imageID{
-		subscriptionID: matches[1],
-		resourceGroup:  matches[2],
-		galleryName:    matches[3],
-		imageName:      matches[4],
-	}, nil
+	return nil
+}
+
+func newGalleryImageVersionsClient(subscriptionID string) (*armcompute.GalleryImageVersionsClient, error) {
```
Comment (r2k1) [inline e2e/config/vhd.go:232]: currently all azure clients are in `config.Azure`.

**Review by r2k1 — APPROVED** [review]
Nice. I've been thinking how to add more replication regions to the packer config.
This seems to be a more elegant solution.

File: e2e/config/vhd.go (hunk 1 above)
Comment (cameronmeissner) [reply e2e/config/vhd.go:232]: ahh good catch, will rework

Comment (cameronmeissner) [reply e2e/config/vhd.go]: fixed

**Review by djsly — APPROVED** [review]

**Pull request merged**